dartagnan/src/main/java/com/dat3m/dartagnan/wmm/relation/base/RelCrit.java

```diff
@@ -23,59 +30,95 @@ public RelCrit(){
     @Override
     public TupleSet getMinTupleSet(){
         if(minTupleSet == null){
-            minTupleSet = new TupleSet();
-            // Todo
+            getMaxTupleSet();
         }
         return minTupleSet;
     }
 
     @Override
     public TupleSet getMaxTupleSet(){
```

Comment:
Since the method populates both min and max sets, it make more sense to extract the populating in a new method which is then call by both `getMinTupleSet()` and `getMaxTupleSet()` whenever one of the sets is `null`.

Reply:
I agree about this. I will fix this